pandas/core/dtypes/base.py

```diff
@@ -233,16 +235,19 @@ def construct_from_string(cls, string: str):
         ...                         "'{}'".format(cls.__name__, string))
         """
         if not isinstance(string, str):
-            raise TypeError("Expects a string, got {}".format(type(string)))
+            raise TypeError("Expects a string, got {typ}".format(typ=type(string)))
```

Comment:
if changing .format could now use f-strings?

Reply:
oh golly, im still tracking down %-based usage

Reply:
I've had a go at upgrading some of our % usage - hopefully there isn't too much more!

https://github.com/pandas-dev/pandas/pull/29518/files